### PR TITLE
feat: add a package that get username on iOS, Android and others

### PR DESCRIPTION
**Pull request by aeddi**
Signed-off-by: aeddi <antoine.e.b@gmail.com>

### CODE REVIEW

File: go/pkg/username/others.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+package username
+
+import "os/user"
+
+const defaultUsername = "anon#1337"
+
+func getUsername() string {
+	user, err := user.Current()
+	if err != nil || user == nil {
+		return ""
```
Comment (moul) [inline go/pkg/username/others.go:12]:
```suggestion
		return defaultUsername
```

File: js/Makefile [hunk 2]
```diff
@@ -16,6 +16,7 @@ APP_CONFIG ?= development
 IOS_APP_TARGET ?= Berty Debug
 ANDROID_APP_TARGET ?= Debug
 IOS_DEVICE ?= iPhone 11
+REQUIRED_JAVA_VER ?= 18
```
Comment (moul) [inline js/Makefile:19]: shouldn't it be hardcoded instead of configurable?

File: go/pkg/username/others.go (hunk 1 above)
Comment (aeddi) [reply go/pkg/username/others.go:12]: Already handled by the public package function `GetUsername()` in go/pkg/username/username.go:12

File: js/Makefile (hunk 2 above)
Comment (aeddi) [reply js/Makefile:19]: Not sure about this, I can hardcode it first and we'll see later?

Comment (moul) [reply js/Makefile]: I'm more interested to have an example of use case where it's a good idea to make this overridable?

**Review by moul — APPROVED** [review]

File: go/pkg/bertymessenger/api.go [hunk 3]
```diff
@@ -85,10 +87,9 @@ func (svc *service) internalInstanceShareableBertyID(ctx context.Context, req *I
 		return nil, errcode.TODO.Wrap(err)
 	}
 
-	displayName := req.DisplayName
+	displayName := strings.TrimSpace(req.DisplayName)
 	if displayName == "" {
-		// FIXME: get it from somewhere
-		displayName = "anonymous#1337"
+		return nil, errcode.TODO
```
Comment (n0izn0iz) [inline go/pkg/bertymessenger/api.go:92]: maybe don't return an error since a client might want to not use a name and the links format supports it

**Review by n0izn0iz — APPROVED** [review]
LGTM, we'll do front in another PR

**Pull request merged**